src/core/a-mixin.js

```diff
@@ -16,23 +33,38 @@ module.exports = registerElement('a-mixin', {
 
       setAttribute: {
         value: function (attr, value) {
-          var component = AComponents[attr];
+          var component = components[attr];
+          var valueString = value;
+          this.attrs[attr] = styleParser.parse(value);
           if (component && typeof value === 'object') {
-            value = component.stringify(value);
+            valueString = component.stringify(value);
           }
-          HTMLElement.prototype.setAttribute.call(this, attr, value);
+          HTMLElement.prototype.setAttribute.call(this, attr, valueString);
         },
         writable: window.debug
       },
 
       getAttribute: {
         value: function (attr) {
-          var component = AComponents[attr];
-          var value = HTMLElement.prototype.getAttribute.call(this, attr);
-          if (!component || typeof value !== 'string') { return value; }
-          return component.parse(value);
+          return this.attrs[attr] || HTMLElement.prototype.getAttribute.call(this, attr);
         },
         writable: window.debug
+      },
+
+      /**
+       * Update cache of parsed component attributes
+       *
+       */
+      updateAttrs: {
+        value: function () {
+          var attributes = this.attributes;
+          var attrName;
+          var i;
+          for (i = 0; i < attributes.length; ++i) {
```

Comment:
Could save a line with just `var i = 0` and still have all variables declared on top